cmd/scheduler-estimator/app/scheduler-estimator.go

```diff
@@ -42,9 +46,22 @@ func NewSchedulerEstimatorCommand(ctx context.Context) *cobra.Command {
 		},
 	}
 
-	opts.AddFlags(cmd.Flags())
+	fss := cliflag.NamedFlagSets{}
+
+	genericFlagSet := fss.FlagSet("generic")
+	genericFlagSet.AddGoFlagSet(flag.CommandLine)
```

Comment:
> I find it in vendor sigs.k8s.io/controller-runtime/pkg/client/config/config.go
> ```
> func init() {
>	// TODO: Fix this to allow double vendoring this library but still register flags on behalf of users
>	flag.StringVar(&kubeconfig, "kubeconfig", "",
>		"Paths to a kubeconfig. Only required if out-of-cluster.")
> }
> ```

Do we need to this default value? If not, we don't need to set `genericFlagSet.AddGoFlagSet(flag.CommandLine)`. @Garrybest

Reply:
The `scheduler-estimator` does not use `controller-runtime` to build. So this flag has nothing to do with it.

Reply:
My question is why the `--kubeconfig` flag will be added by controller-runtime, seems `scheduler estimater` didn't import controller-runtime?

Reply:
```
func run(ctx context.Context, opts *options.Options) error {
	klog.Infof("karmada-scheduler-estimator version: %s", version.Get())
	go serveHealthzAndMetrics(net.JoinHostPort(opts.BindAddress, strconv.Itoa(opts.SecurePort)))

	restConfig, err := clientcmd.BuildConfigFromFlags(opts.Master, opts.KubeConfig)
       ...
```

Reply:
I guess it is imported from other package that `scheduler-estimator` invokes.

Reply:
That's possible. But, where?

Reply:
I am also very curious.

Reply:
I have modified it correctly

Reply:
Ok, thank you. @AllenZMC